crates/aptos-crypto/src/bls12381/bls12381_keys.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides APIs for private keys and public keys used in BLS multi-signatures
+//! implemented on top of BLS12-381 elliptic curves (https://github.com/supranational/blst).
+//!
+//! The `PublicKey` struct is used to represent both the public key of an individual signer
+//! as well as the aggregate public key of several signers. Before passing such a struct as an
+//! argument the caller should always ensure it has verified its proof-of-possession via
+//! `ProofOfPossession::verify`.
+//!
+//! The `PublicKey::aggregate` API assumes the caller has already verified
+//! proofs-of-possession for all the given public keys and therefore all public keys are valid group
+//! elements.
+//!
+//! In general, with the exception of `ProofOfPossession::verify` no library function should
+//! be given a public key as argument without first verifying that public key's PoP. Note that
```

Comment:
I suggest you put "(PoP)" next to "proofs of possession", as in "proofs of possession (PoP)" somewhere above in these docstrings, so that subsequent mentions of "PoP" are clear. It is clear enough even without specifying, but we don't want the reader to guess.